AshAvalanche/ansible-avalanche-collection
# feat: add faucet role

### Linked issues

- Fix #36 

### Changes

- Add the `ash.avalanche.faucet` role and `install_faucet_docker` playbook

### Additional comments

A few remarks:
- The faucet is running on Docker with the `ghcr.io/ashavalanche/faucet-image:disable-captcha-variable` built from our [`disable-captcha-variable`](https://github.com/AshAvalanche/avalanche-faucet/tree/disable-captcha-variable) branch.
- The way Ava Labs did things, `config.json` is included in the image which is not really flexible. For now we override it with a volume in the docker-compose file to make a configurable container instance.
- The `erc20tokens` configuration allows to load additional ERC-20 tokens in the faucet. Referring to a something that is not an actual deployed ERC-20 smart contract makes the app fail so let's not support this for the first iteration.
- The `evmchains` supports multiple chains. We only template a single chain in the config file for now to keep things simple at first.

### How to test

- Add `faucet` group with `validator01`
- Override `avalanche_faucet_config['subnet_rpc']` with a valid RPC
- Run `ansible-playbook ash.avalanche.install_faucet_docker -i inventories/local`
- The faucet should be available at http://192.168.60.11:8000/

I will write the README and tutorial for Ash docs once you validate the technical part.
